### PR TITLE
fix: pass connection state to activity emoji

### PR DESCRIPTION
**Pull request by shiftinv**
## Summary

Fixes an issue where you couldn't fetch the emoji data of an `Activity`/`CustomActivity` due to a missing `ConnectionState`.

## Checklist

- [x] If code changes were made, then they have been tested
    - [ ] I have updated the documentation to reflect the changes
    - [x] I have formatted the code properly by running `task lint`
    - [x] I have type-checked the code by running `task pyright`
- [x] This PR fixes an issue
- [ ] This PR adds something new (e.g. new method or parameters)
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

**Comment by onerandomusername** [comment]
This feels overcomplicated. Is it possible to fix this just by setting the state in the specific internal location where custom activity is created?

However, is this approach better than the one taken in this pull? At this point in time there is no reason for custom activities to be user creatable, but discord could always change that. However, in that case emoji would not need to have a state injected.

**Comment by shiftinv** [comment]
Some of it could be avoided by instead setting `activity.emoji._state` in `create_activity` since all the other calls go through that; it'd end up working mostly the same way, but might look better (?)

**Comment by shiftinv** [comment]
> Is it possible to fix this just by setting the state in the specific internal location where custom activity is created?

Alright, updated the code accordingly, and cleaned up `create_activity` a bit in the process.

**Review by onerandomusername — CHANGES_REQUESTED** [review]
just a code review right now

File: disnake/activity.py [hunk 1]
```diff
@@ -776,7 +784,7 @@ def __init__(self, name: Optional[str], *, emoji: Optional[PartialEmoji] = None,
             self.emoji = emoji
         else:
             raise TypeError(
-                f"Expected str, PartialEmoji, or None, received {type(emoji)!r} instead."
+                f"Expected dict, str, PartialEmoji, or None, received {type(emoji)!r} instead."
```
[the remark this answers is not in the document]
Comment (shiftinv) [reply disnake/activity.py]: 3bdb64ad22ebeb66eee1a3f400df72c47a1e3b86

**Review by onerandomusername — APPROVED** [review]
looks good to me, tested, and it seems to work properly.

**Pull request merged**